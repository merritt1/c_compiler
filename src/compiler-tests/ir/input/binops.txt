int main(void) {
	int i, j, k, l;
	i=j+k*l;
}