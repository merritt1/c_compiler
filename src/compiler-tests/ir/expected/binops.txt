(procBegin,  main)
(addressOf,     r0,      i)
(addressOf,     r1,      j)
(addressOf,     r2,      k)
(addressOf,     r3,      l)
(loadWord,     r4,      r2)
(loadWord,     r5,      r3)
(multSignedWord,     r6,      r4,      r5)
(loadWord,     r7,      r1)
(addSignedWord,     r8,      r7,      r6)
(storeWord,     r0,      r8)
(procEnd ,  main)
